core/scheduler/scheduler.go

```diff
@@ -170,6 +172,16 @@ func (s *Scheduler) GetDutyDefinition(ctx context.Context, duty core.Duty) (core
 	}
 
 	epoch := duty.Slot / slotsPerEpoch
+
+	// This has to be very rare event, when the requested epoch is being resolved.
+	// We wait for the epoch to be resolved before returning the duty definition.
+	for s.isResolvingEpoch(epoch) {
+		if ctx.Err() != nil {
+			return nil, errors.Wrap(ctx.Err(), "context cancelled while waiting for epoch to resolve")
+		}
+		time.Sleep(100 * time.Millisecond)
```

Comment:
Probably will be better if we wait on some channel, but not sure if it's worth the time implementing that?

Reply:
I was thinking about the same first, but we need to compare against specific epoch number.. which adds lots of complexity when done via channels. Because this is an edge case, I think it is sufficient to do just busy-wait.